src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -371,24 +371,24 @@ protected void CheckDocumentIsOpen(DocumentId documentId)
                 AddToOpenDocumentMap(documentId);
```

Comment:
these methods do massive amounts of stateful changes.  def cannot use the SetCurrentSolution helper currently.